File: page.ts
class Student {

    id: number;
    name: string;
    age: number;
    marks: number;


    constructor(
        id: number,
        name: string,
        age: number,
        marks: number
    ) {
        this.id = id;
        this.name = name;
        this.age = age;
        this.marks = marks;
    }


    display(): void {

        console.log("---------------------");
        console.log("ID:", this.id);
        console.log("Name:", this.name);
        console.log("Age:", this.age);
        console.log("Marks:", this.marks);

    }
}



let students: Student[] = [];



function addStudent(
    id: number,
    name: string,
    age: number,
    marks: number
): void {

    let student = new Student(
        id,
        name,
        age,
        marks
    );


    students.push(student);


    console.log(
        "Student added successfully"
    );
}




function displayStudents(): void {

    console.log(
        "Student Records"
    );


    if (students.length === 0) {

        console.log(
            "No students found"
        );

        return;
    }


    for (let student of students) {

        student.display();

    }

}





function searchStudent(
    id: number
): void {


    let student = students.find(
        s => s.id === id
    );


    if (student) {

        console.log(
            "Student Found"
        );

        student.display();

    }
    else {

        console.log(
            "Student not found"
        );

    }

}





function deleteStudent(
    id: number
): void {


    let index = students.findIndex(
        s => s.id === id
    );


    if (index !== -1) {

        students.splice(
            index,
            1
        );


        console.log(
            "Student deleted"
        );

    }
    else {

        console.log(
            "Student not found"
        );

    }

}





function calculateAverage(): number {


    if (students.length === 0) {

        return 0;

    }


    let total: number = 0;


    students.forEach(
        student => {

            total += student.marks;

        }
    );


    return total / students.length;

}





function main(): void {


    addStudent(
        1,
        "John",
        20,
        85
    );


    addStudent(
        2,
        "Mary",
        21,
        90
    );


    addStudent(
        3,
        "David",
        19,
        75
    );



    displayStudents();



    searchStudent(2);



    deleteStudent(1);



    console.log(
        "Average Marks:",
        calculateAverage()
    );



    displayStudents();


}



main();

File: page.pas
program StudentManagement;

uses crt;

const
  MAX = 50;

type
  Student = record
    id: integer;
    name: string[50];
    age: integer;
    marks: real;
  end;

var
  students: array[1..MAX] of Student;
  count: integer;

procedure AddStudent;
begin
  count := count + 1;
  writeln('Enter student ID:');
  readln(students[count].id);

  writeln('Enter student name:');
  readln(students[count].name);

  writeln('Enter age:');
  readln(students[count].age);

  writeln('Enter marks:');
  readln(students[count].marks);

  writeln('Student added successfully');
end;

procedure DisplayStudents;
var
  i: integer;
begin
  writeln('Student List');
  writeln('------------');

  for i := 1 to count do
  begin
    writeln('ID: ', students[i].id);
    writeln('Name: ', students[i].name);
    writeln('Age: ', students[i].age);
    writeln('Marks: ', students[i].marks:0:2);
    writeln;
  end;
end;

procedure SearchStudent;
var
  searchID, i: integer;
begin
  writeln('Enter ID to search:');
  readln(searchID);

  for i := 1 to count do
  begin
    if students[i].id = searchID then
    begin
      writeln('Student found:');
      writeln(students[i].name);
      exit;
    end;
  end;

  writeln('Student not found');
end;

var
  choice: integer;

begin
  clrscr;
  count := 0;

  repeat
    writeln('1. Add Student');
    writeln('2. Display Students');
    writeln('3. Search Student');
    writeln('4. Exit');

    readln(choice);

    case choice of
      1: AddStudent;
      2: DisplayStudents;
      3: SearchStudent;
    end;

  until choice = 78;

  writeln('Program ended');
  readln;
end.
